_posts/2018-09-18-How-to-build-a-Serverless-Application-with-AWS-Lambda-and-DynamoDB.md

```diff
@@ -0,0 +1,563 @@
+* `docClient.scan(scanningParameters, function(err,data)` executes the scan and returns either the result or the error that occurred.
+
+## Test write-Lambda
+All right! Let's test this thing..
+On the Lambda Function configuration page you see a dropdown and test button in the upper right corner.
+Click the dropdown and configure a new test event.
+I called mine 'Test' and added an empty test event.
+
+<div style="text-align: center;">
+  <img src="/img/2018-09-18-How-to-Build-a-Serverless-Application/Lambda-CodingTips_Scan-Created_Empty_Test_Event.png" width="60%" height="60%">
+</div>  
+
+**Save** it and you are ready to test the Lambda.  
+From the dropdown select your test event and hit the **Test** button!
+Nice one, this returns the items in your table:
+
+<div style="text-align: center;">
+  <img src="/img/2018-09-18-How-to-Build-a-Serverless-Application/Lambda-CodingTips_Scan-Execution_Result_Succeeded.png" width="100%">
+</div>
+
+# API Gateway: Access the scan Lambda
+Mmmh, fine.. We can trigger the Lambda Function with a test event.
+But we want to be able to trigger it from anywhere using a URL.
+In the designer view of the lambda you can still see **add triggers from the left**.
+Well, let's add that trigger!
+To expose a Lambda Function AWS provides the **API Gateway**.
+
+Under **Services** navigate to **API Gateway**. 
+> Amazon API Gateway is a fully managed service that makes it easy for developers to create, publish, maintain, monitor, and secure APIs at any scale. 
+With a few clicks in the AWS Management Console, you can create an API that acts as a “front door” for applications -- from AWS docs: <a target="_blank" href="https://aws.amazon.com/api-gateway/">https://aws.amazon.com/api-gateway</a>
+
+Basically this is the Service you use to create all of your API's.  
+* Click **Create api** and name your api **CodingTips**
+* Add a description if you like
+* Leave the **Endpoint Type** to regional and **create API**
+
+<div style="text-align: center;">
+  <img src="/img/2018-09-18-How-to-Build-a-Serverless-Application/API_Gateway-GET-Create_API.png" width="100%">
+</div>
+ 
+The API has been created.
+Configure it by adding a resource.
+
+* Under **Actions** click **Create Resource** and name it **tips** with **/tips** as **Resource Path**
+* Check **Enable CORS** to make your API accessible from anywhere
+* Hit **Create Resource**
+
+Time to configure the HTTP GET request.
+
+* Select the **/tips** endpoint
+* Under **Actions** create the **GET** method
+* Integration type is **Lambda Function**
+* As Lambda Function provide the name of the lambda.
+In this case that is Codingtips_Scan.
+* Save the configuration.
+
+<div style="text-align: center;">
+  <img src="/img/2018-09-18-How-to-Build-a-Serverless-Application/API_Gateway-GET-Configuration.png" width="100%">
+</div>
+
+Only one thing left: select the api and under **Actions** click **Deploy API**. 
```

Comment:
API

Reply:
I did set the text to "Deploy API" (uppercase) because this is how the text is shown on the button in the AWS console

Reply:
Talking about the lowercase api. :)

Reply:
okay, got it.